controllers/storagecluster/external_resources.go

```diff
@@ -299,7 +300,27 @@ func (obj *ocsExternalResources) ensureDeleted(r *StorageClusterReconciler, inst
 		}
 		defer externalClusterClient.Close()
 
-		return r.offboardConsumer(instance, externalClusterClient)
+		if res, err := r.offboardConsumer(instance, externalClusterClient); err != nil {
+			return res, err
+		}
+
+		cephFilesystemSubVolumeGroupList := &cephv1.CephFilesystemSubVolumeGroupList{}
+
+		if err = r.Client.List(context.TODO(), cephFilesystemSubVolumeGroupList, client.InNamespace(instance.Namespace)); err != nil {
+			return reconcile.Result{}, fmt.Errorf("uninstall: Failed to fetch cephFilesystemSubVolumeGroupList. %v", err)
+		}
+
+		for _, cephFilesystemSubVolumeGroup := range cephFilesystemSubVolumeGroupList.Items {
+			if err = r.Client.Delete(context.TODO(), &cephFilesystemSubVolumeGroup); err != nil && !errors.IsNotFound(err) {
+				r.Log.Error(
+					err,
+					"Uninstall: Failed to delete CephFilesystemSubVolumeGroup.",
+					"CephFilesystemSubVolumeGroup",
+					klog.KRef(cephFilesystemSubVolumeGroup.Namespace, cephFilesystemSubVolumeGroup.Name),
+				)
+				return reconcile.Result{}, fmt.Errorf("uninstall: Failed to delete CephFilesystemSubVolumeGroup %q. %v", cephFilesystemSubVolumeGroup.Name, err)
```

Comment:
@kesavanvt 
Do we really need an error here? can we do a warning and keep deleting any other resources?
What will happen if we will keep the resource?

Reply:
AFAIF, this error will be raised in case of any Kubernetes issue. I think its better to return an error rather than just logging the warning. If we keep the resource, the cephcluster deletion will get stuck .

Reply:
Ok, so what about keeping an error flag outside the loop. 
Then for each resource that cannot be deleted add a warning and set the flag to true.
At the end of the loop if the flag is on, return a general error about being unable to delete resources.

WDYT?